app/components/article-content.js

```diff
@@ -607,67 +608,66 @@ export default Component.extend(
 
     /**
      * Injects an affiliate unit into the article content
-     * @param {string} title
      */
-    handleBigAffiliateUnit(title) {
-      const unit = this.affiliateSlots.getBigUnitOnPage(title);
-
-      if (typeof unit === 'undefined') {
-        // There's no unit to display (not an error)
-        return;
-      }
-
-      // search for second section
-      const h2Elements = this.element.querySelectorAll('h2[section]');
+    handleBigAffiliateUnit() {
+      this.affiliateSlots
+        .fetchUnitForPage(this.id, true)
+        .then(unit => {
+          if (typeof unit === 'undefined') {
+            // There's no unit to display (not an error)
+            return;
+          }
 
-      if (h2Elements[1]) {
-        const unitPlaceholder = document.createElement('div');
-        const unitWrapper = document.createElement('div');
+          // search for second section
+          const h2Elements = this.element.querySelectorAll('h2[section]');
```

Comment:
This might be a little late in the game now but are we certain that we only want to base this on `h2`? How normalised is the generated content that we can be somewhat confident this will show where we want it? This is what worries me the most as we can see when enabled across all pages we do run into some weird issues.